core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -208,10 +208,23 @@ object BattleDamage {
 
     private fun getTileSpecificModifiers(unit: MapUnitCombatant, tile: TileInfo): HashMap<String,Float> {
         val modifiers = HashMap<String,Float>()
-        if(tile.isFriendlyTerritory(unit.getCivInfo()) && unit.getCivInfo().hasUnique("+15% combat strength for units fighting in friendly territory"))
-            modifiers["Himeji Castle"] = 0.15f
-        if(!tile.isFriendlyTerritory(unit.getCivInfo()) && unit.unit.hasUnique("+20% bonus outside friendly territory"))
-            modifiers["Foreign Land"] = 0.2f
+
+        if(tile.isFriendlyTerritory(unit.getCivInfo())) {
+            if (unit.getCivInfo().hasUnique("+15% combat strength for units fighting in friendly territory")) {
+                modifiers["Himeji Castle"] = 0.15f
+            }
+            for(unique in unit.getCivInfo().getMatchingUniques("+[]% combat strength for units fighting in friendly territory")) {
+                modifiers["[${unique.params[0]}] nation bonus in friendly territory"] = unique.params[0].toFloat() / 100
```

Comment:
The modifier texts need to be in template.properties

Reply:
Thank you to you and HadeanLake for assistance.

-Added a function that allows for stacked percent modifiers.
-Bonus in own territory consolidated as "Friendly Land" modifier. Can stack with multiple of this same parameterized unique. Old way still works for now, but added comment that it is deprecated.
-Bonus in foreign territory consolidated as "Foreign Land" modifier.  Can stack with multiple of this same parameterized unique. Old way still works for now, but added comment that it is deprecated.
-Added "Friendly Land" to templates.properties for translation
-Modified units and building json files to change the uniques on Himeji Castle and Foreign Legion to use the new way of doing these uniques.